### PR TITLE
Add features from next to master

### PR DESCRIPTION
**Pull request by gilligan**
As described in #27 I rebased next fixing two failing tests. I also updated the README describing the new record syntax and the possibility to define instance methods.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,6 +44,32 @@ Thus the above example is equivalent to this:
...
+var Point = Type({Point: {x: Number, y: Number}});
+```
+
+Furthermore it is possible to add instance methods. A Maybe type with a map
+function could thus be defined as follows:
+
+```javascript
+var T = function () { return true; };
+var maybeMap(fn) {
+    var this = this;
```
Comment (svozza) [inline README.md]: Should the variable not be called `self` or something like that rather than `this`?

**Comment by gilligan** [comment]
@paldepind care to have a look at this? :)

**Comment by paldepind** [comment]
Hello @gilligan. I am sorry for leaving you waiting. I just had my last exam today—so I have more time now.

Thank you a lot_ for bringing the changes in the `next` branch up to date and for pushing me to get these changes published.

This is my current plan:

* Merge this PR
* Fix the problem I mentioned here:
> The biggest problem in my implementation is that when you create a value from an object the fields will be put both in numerical properties (0, 1, etc.) and in the field name properties (name, id, etc.). I think not keeping numerical properties for record-like-types is the solution.

* Implement an idea for a better way to define instance methods.
* Open a new issue to hear what people think

Does that sound fine to you?

**Comment by paldepind** [comment]
Is it possible to merge this against the `next` branch instead of the `master` branch?

**Comment by gilligan** [comment]
Sure, I will do a new PR tomorrow.

**Comment by paldepind** [comment]
@gilligan Thank you :smile:

**Pull request closed without merging**